### PR TITLE
Unordered edits allowed

### PR DESCRIPTION
**Pull request by bbridge0200**
1. done: remove most references to ordering of images in csv (except for when given a new shuffled order which is then written into the csv) 
2. done: allow csv uploaded image sets to be shuffled, added to, and deleted from

### CODE REVIEW

File: napari_allencell_annotator/controller/main_controller.py [hunk 1]
```diff
@@ -122,27 +230,157 @@ def _start_annotating_clicked(self):
...
+            self.annots.start_annotating(self.images.get_num_files(), dct, shuffled)
         self.annots.set_curr_img(self.images.curr_img_dict())
+        if not shuffled:
+            self.images.view.file_widget.currentItemChanged.connect(self.image_selected)
+            self.images.view.input_dir.hide()
+            self.images.view.input_file.hide()
+            self.images.view.shuffle.hide()
+            self.images.view.delete.hide()
+
+    def fix_already_annotated(self, dct):
```
Comment (aditnath) [inline napari_allencell_annotator/controller/main_controller.py]: Just add comments here describing functionality of these functions

**Review by aditnath — APPROVED** [review]
Looks great, looks like you took out the order for images, which will help with future features too

File: napari_allencell_annotator/assets/test.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Shuffled:,FALSE,,,,
+Annotations:,"{""name"": {}, ""name2"": {}, ""name3"": {}}",,,,
```
Comment (yrkim98) [inline napari_allencell_annotator/assets/test.csv]: if these are for testing can you put them into _tests/assets
if not just remove them

File: napari_allencell_annotator/controller/annotator_controller.py [hunk 3]
```diff
@@ -43,40 +45,65 @@ class AnnotatorController:
...
-        # path = str(Directories.get_assets_dir() / "sample3.json")
-        # 4 annotations
-        # path = str(Directories.get_assets_dir() / "sample.json")
-        # 8 annotations
-        # path: str = str(Directories.get_assets_dir() / "sample2.json")
-        # self.annot_json_data: Dict[str, Dict] = json.load(open(path))
 
         # dictionary of json info:
-        self.annot_json_data: Dict[str, Dict] = None
+        self.annot_json_data: Dict[str, Dict[str, Union[int, str, bool, List[str]]]] = None
```
Comment (yrkim98) [inline napari_allencell_annotator/controller/annotator_controller.py]: might as well just use the `Any` type

File: napari_allencell_annotator/controller/annotator_controller.py [hunk 4]
```diff
@@ -43,40 +45,65 @@ class AnnotatorController:
...
+
+    def write_json(self, file_path: str):
+        """
+        Write annotation dictionary to a file.
+
+        file_path : str
+            file path for json file to write to.
+        """
+        if self.annot_json_data is not None:
+            json.dump(self.annot_json_data, open(file_path, "w"), indent=4)
```
Comment (yrkim98) [inline napari_allencell_annotator/controller/annotator_controller.py:106]: have you tested that this works

File: napari_allencell_annotator/controller/annotator_controller.py [hunk 5]
```diff
@@ -145,9 +172,12 @@ def set_curr_img(self, curr_img: Optional[Dict[str, str]] = None):
             # if at the end disable next
             if int(curr_img["Row"]) == self.view.num_images - 1:
```
Comment (yrkim98) [inline napari_allencell_annotator/controller/annotator_controller.py]: could you clean this if/else up?
I feel like you could do this
```
if int(curr_img["Row"]) == self.view.num_images - 1 or int(curr_img["Row"]) == 0:
self.view.next_btn.setEnabled(False)
else:
True
```

File: napari_allencell_annotator/view/annotator_view.py [hunk 6]
```diff
@@ -177,25 +182,46 @@ def mode(self) -> AnnotatorViewMode:
...
             self.progress_bar.setText("{} of {} Images".format(self.curr_index + 1, self.num_images))
 
     def _reset_annotations(self):
         """Reset annotation data to empty."""
         self.annot_list.clear()
-        # todo
+        # todo: make size policy, remove this line
         self.annot_list.setMaximumHeight(600)
-        self.annotation_item_widgets: List[QWidget] = []
+        self.annotation_item_widgets = []
```
Comment (yrkim98) [inline napari_allencell_annotator/view/annotator_view.py]: Type for this?

**Review by yrkim98 — CHANGES_REQUESTED** [review]
LGTM! Just make sure you get lint/build to pass before merging.

File: napari_allencell_annotator/controller/annotator_controller.py (hunk 5 above)
Comment (bbridge0200) [reply napari_allencell_annotator/controller/annotator_controller.py]: I think I need both, if row is len-1 you enable the next button whereas if len  is 0 you enable the previous button so I would need another if statement anyways

**Pull request merged**